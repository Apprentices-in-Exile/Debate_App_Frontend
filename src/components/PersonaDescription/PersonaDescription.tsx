import { Avatar, Typography } from '@material-tailwind/react'

interface PersonaDescriptionProps {
  persona: {
    imageURL: string
    name: string
    description: string
  }
}

function PersonaDescription ({ persona }: PersonaDescriptionProps): JSX.Element {
  const { name, imageURL, description } = persona
  console.log('stuff')
  return (
    <div className="flex items-center gap-4">
      <Avatar src={imageURL} alt="avatar" />
      <div>
        <Typography variant="lead">{name}</Typography>
        <Typography variant="small" color="gray" className="font-normal">{description}</Typography>
      </div>
    </div>
  )
}

export default PersonaDescription
