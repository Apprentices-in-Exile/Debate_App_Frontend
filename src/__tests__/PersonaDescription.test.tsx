import renderer from 'react-test-renderer'
import PersonaDescription from '../components/PersonaDescription/PersonaDescription'

describe('PersonaDescription', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<PersonaDescription persona={{
      imageURL: 'imageURL',
      name: 'Name',
      description: 'Description'
    }} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
