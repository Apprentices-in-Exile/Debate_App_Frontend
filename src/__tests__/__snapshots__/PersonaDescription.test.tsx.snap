// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonaDescription should match the snapshot 1`] = `
<div
  className="flex items-center gap-4"
>
  <img
    alt="avatar"
    className="inline-block relative object-cover object-center !rounded-full w-12 h-12 rounded-lg"
    src="imageURL"
  />
  <div>
    <p
      className="block antialiased font-sans text-xl font-normal leading-relaxed text-inherit"
    >
      Name
    </p>
    <p
      className="block antialiased font-sans text-sm leading-normal text-gray-700 font-normal"
    >
      Description
    </p>
  </div>
</div>
`;
